substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1528,40 +1530,43 @@ boolean useBundle() {
         for (String affectingEnvironmentVariable : jvmAffectingEnvironmentVariables) {
             environment.remove(affectingEnvironmentVariable);
         }
     }
 
     private static void sanitizeJVMEnvironment(Map<String, String> environment, Map<String, String> imageBuilderEnvironment) {
-        Map<String, String> restrictedEnvironment = new HashMap<>();
-        List<String> jvmRequiredEnvironmentVariables = new ArrayList<>(List.of("PATH", "PWD", "HOME", "LANG", "LC_ALL"));
-        jvmRequiredEnvironmentVariables.add("SRCHOME"); // FIXME
+        Set<String> requiredKeys = new HashSet<>(List.of("PATH", "PWD", "HOME", "LANG", "LC_ALL"));
+        requiredKeys.add("SRCHOME"); /* Remove once GR-44676 is fixed */
+        Function<String, String> keyMapper;
         if (OS.WINDOWS.isCurrent()) {
-            jvmRequiredEnvironmentVariables.addAll(List.of("TEMP", "INCLUDE", "LIB"));
+            requiredKeys.addAll(List.of("TEMP", "INCLUDE", "LIB"));
+            keyMapper = String::toUpperCase;
+        } else {
+            keyMapper = Function.identity();
         }
-        for (String requiredEnvironmentVariable : jvmRequiredEnvironmentVariables) {
-            String val = environment.get(requiredEnvironmentVariable);
-            if (val != null) {
-                restrictedEnvironment.put(requiredEnvironmentVariable, val);
+        Map<String, String> restrictedEnvironment = new HashMap<>();
+        environment.forEach((key, val) -> {
+            if (requiredKeys.contains(keyMapper.apply(key))) {
+                restrictedEnvironment.put(key, val);
             }
-        }
+        });
         for (Iterator<Map.Entry<String, String>> iterator = imageBuilderEnvironment.entrySet().iterator(); iterator.hasNext();) {
             Map.Entry<String, String> entry = iterator.next();
-            String requiredKey = entry.getKey();
-            String requiredValue = entry.getValue();
-            if (requiredValue != null) {
-                restrictedEnvironment.put(requiredKey, requiredValue);
+            if (entry.getValue() != null) {
+                restrictedEnvironment.put(entry.getKey(), entry.getValue());
             } else {
-                String existingValue = environment.get(requiredKey);
-                if (existingValue != null) {
-                    restrictedEnvironment.put(requiredKey, existingValue);
-                    /* Capture found existingValue for storing vars in bundle */
-                    entry.setValue(existingValue);
-                } else {
-                    NativeImage.showWarning("Environment variable '" + requiredKey + "' is undefined and therefore not available during image build-time.");
+                environment.forEach((key, val) -> {
+                    if (keyMapper.apply(key).equals(keyMapper.apply(entry.getKey()))) {
+                        restrictedEnvironment.put(entry.getKey(), val);
```

Comment:
I think it would be better to use the key from the actual environment instead of the key passed to `-E`.
```suggestion
                        restrictedEnvironment.put(key, val);
```

Reply:
I did this on purpose. For Windows it does not mapper if the key has upper/lowercase differences. But for Linux it matters. Recording the key as it was given by `-E<key-name>` allows users to create bundles on Windows that will also work on Linux. Using the key from the actual environment cannot not guarantee that the env-setting will be usable on Linux. For example:

Windows environment contains `foo=bar` and Java code uses `System.getEnv("FOO")` at image build-time and everything seems to work on Windows.
 
* Now if I use `-EFOO` and I would persist the env-setting as `foo=bar` in the bundle, building the image on Linux would not work.
* But if I persist it instead as ` FOO=bar` the same bundle would also build correctly on Linux.

Reply:
:+1:

It's probably worth documenting this in the code for future reference.

Reply:
OK. I will add a comment in a follow up PR.

Reply:
See https://github.com/oracle/graal/pull/6243/commits/706b2febb44f37cae454ab4d7c69de3aca755c01

Reply:
LGTM, thanks Paul.